store/job_store_test.go

```diff
@@ -0,0 +1,101 @@
+			So(job.NumberOfTasks, ShouldEqual, expectedJob.NumberOfTasks)
+			So(job.ReindexCompleted, ShouldEqual, expectedJob.ReindexCompleted)
+			So(job.ReindexFailed, ShouldEqual, expectedJob.ReindexFailed)
+			So(job.ReindexStarted, ShouldEqual, expectedJob.ReindexStarted)
+			So(job.SearchIndexName, ShouldEqual, expectedJob.SearchIndexName)
+			So(job.State, ShouldEqual, expectedJob.State)
+			So(job.TotalSearchDocuments, ShouldEqual, expectedJob.TotalSearchDocuments)
+			So(job.TotalInsertedSearchDocuments, ShouldEqual, expectedJob.TotalInsertedSearchDocuments)
+
+			Convey("Return with error when the same job id is used a second time", func() {
+				expectedErrorMsg := "id must be unique"
+				job, err := jobStore.CreateJob(ctx, inputID)
+
+				So(err, ShouldNotBeNil)
+				So(job, ShouldResemble, models.Job{})
+				So(err.Error(), ShouldEqual, expectedErrorMsg)
+			})
+		})
+	})
+	Convey("Return with error when the job id is an empty string", t, func() {
+		inputID := ""
+		expectedErrorMsg := "id must not be an empty string"
+		jobStore := DataStore{}
+		job, err := jobStore.CreateJob(ctx, inputID)
+
+		So(job, ShouldResemble, models.Job{})
+		So(err, ShouldNotBeNil)
+		So(err.Error(), ShouldEqual, expectedErrorMsg)
+	})
+}
+
+//TestGetJob tests the GetJob function in job_store.
+func TestGetJob(t *testing.T) {
+	Convey("Successfully return without any errors", t, func() {
+		Convey("when the job id exists in the jobStore", func() {
+			inputID := testJobID2
+			jobStore := DataStore{}
+
+			//first create a job so that it exists in the job store
+			job, err := jobStore.CreateJob(ctx, inputID)
+			So(err, ShouldBeNil)
+
+			//check that the job created is not an empty job
+			So(job, ShouldNotResemble, models.Job{})
+
+			//then get the job and check that it contains the same values as the one that's just been created
+			job_returned, err := jobStore.GetJob(ctx, inputID)
+			So(err, ShouldBeNil)
+			So(job.ID, ShouldEqual, job_returned.ID)
+			So(job.Links, ShouldResemble, job_returned.Links)
+			So(job.NumberOfTasks, ShouldEqual, job_returned.NumberOfTasks)
+			So(job.ReindexCompleted, ShouldEqual, job_returned.ReindexCompleted)
+			So(job.ReindexFailed, ShouldEqual, job_returned.ReindexFailed)
+			So(job.ReindexStarted, ShouldEqual, job_returned.ReindexStarted)
+			So(job.SearchIndexName, ShouldEqual, job_returned.SearchIndexName)
+			So(job.State, ShouldEqual, job_returned.State)
+			So(job.TotalSearchDocuments, ShouldEqual, job_returned.TotalSearchDocuments)
+			So(job.TotalInsertedSearchDocuments, ShouldEqual, job_returned.TotalInsertedSearchDocuments)
+		})
+	})
+	Convey("Return with error when the job id does not exist in the jobStore", t, func() {
+		inputID := testJobID3
+		expectedErrorMsg := "the job store does not contain the job id entered"
+		jobStore := DataStore{}
+		job, err := jobStore.GetJob(ctx, inputID)
+
+		So(err, ShouldNotBeNil)
+		So(job, ShouldResemble, models.Job{})
+		So(err.Error(), ShouldEqual, expectedErrorMsg)
```

Comment:
Think about adding the following two tests:

1) Failure when the job id is empty
2) Failure when the jobStore map is nil

Reply:
I've had a think but I don't think that either of those situations can really occur, here's why:
1. If the job id is empty then the endpoint call is just to jobs/ and that gives "404 page not found", which I guess could be tested but it's due to be implemented in a later ticket - so it will then return a list of jobs instead of an error.
2. The jobStore map gets made when the DataStore is created, as far as I can tell, and so it's always there I think. I previously had some code to create it when the first job was created but I took that out a few days ago as it didn't seem to be needed.

Reply:
I only mention it because there is logic in the jobStore that checks for the id and checks if the map is nil.

So even it is not possible to have a request without the id from the API interface level, the jobStore does not know that and a different method/handler may call it and pass and empty string for an id at which point it should fail. Hence we should be adding a unit test for this in the store package.